sim/pokemon.ts

```diff
@@ -684,9 +700,7 @@ export class Pokemon {
 		return (lockedMove === true) ? null : lockedMove;
 	}
 
-	getMoves(lockedMove?: string | null, restrictData?: boolean):
-	{move: string, id: string, disabled?: string | boolean,
-		disabledSource?: string, target?: string, pp?: number, maxpp?: number}[] {
+	getMoves(lockedMove?: string | null, restrictData?: boolean): RequestMoveSlot[] {
```

Comment:
`getMoveRequest` is probably the best name here. Request functions have always been a mess.